File: src/main/java/ru/mm/surv/capture/repository/PropertiesWebcamRepository.java
package ru.mm.surv.capture.repository;

import org.springframework.stereotype.Repository;
import ru.mm.surv.capture.config.CameraConfig;
import ru.mm.surv.capture.config.FfmpegConfig;

import java.util.Collection;

@Repository
public class PropertiesWebcamRepository implements WebcamRepository {
    private FfmpegConfig config;

    public PropertiesWebcamRepository(FfmpegConfig config) {
        this.config = config;
    }

    @Override
    public Collection<CameraConfig> getAll() {
        return config.getRecorder().values();
    }
}

File: src/main/java/ru/mm/surv/capture/repository/impl/PropertiesWebcamRepository.kt
package ru.mm.surv.capture.repository.impl

import org.springframework.stereotype.Repository
import ru.mm.surv.capture.config.FfmpegConfig
import ru.mm.surv.capture.repository.WebcamRepository
import ru.mm.surv.capture.config.CameraConfig

@Repository
class PropertiesWebcamRepository(private val config: FfmpegConfig) : WebcamRepository {

    override fun all(): Collection<CameraConfig> {
        return config.recorder.values
    }
}
